opensearch_py_ml/ml_commons/model_execute.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import json
+
+from opensearchpy import OpenSearch
+
+from opensearch_py_ml.ml_commons.ml_common_utils import ML_BASE_URI
+
+
+class ModelExecute:
+    """
+    Class for executing algorithms using ML Commons execute API.
+    """
+
+    API_ENDPOINT = "models/_execute"
+
+    def __init__(self, os_client: OpenSearch):
+        self._client = os_client
+
+    def _execute(self, algorithm_name: str, input_json: dict) -> dict:
+        """
+        This method executes ML algorithms that can be only executed directly (i.e. do not support train and
+        predict APIs), like anomaly localization and metrics correlation.
+        The input json must be a dictionary or a deserializable Python object.
+        The algorithm has to be supported by ML Commons.
+        Refer to https://opensearch.org/docs/2.7/ml-commons-plugin/api/#execute
+
+        :param algorithm_name: Name of the algorithm
+        :type algorithm_name: string
+        :param input_json: Dictionary of parameters or a deserializable string, byte, or bytearray
+        :type input_json: dict
+        :return: returns the API response
+        :rtype: dict
+        """
+
+        if not isinstance(input_json, dict):
```

Comment:
It will return false if the input is dictionary

Reply:
Is that okay?

Reply:
you mean if the input is `{"a":"abc"}` then it will fail?

Reply:
Yeah, I think so. json.loads accepts only string or bytearray

Reply:
I checked using this function:

```
Provided input string is valid:  {"a":"abc"}
Provided input string is valid:  {"operation": "max", "input_data": [1.0, 2.0, 3.0]}
Provided input string is not valid:  zaa
Provided input string is valid:  {"index_name": "rca-index","attribute_field_names": ["attribute"],"aggregations": [{"sum": {"sum": {"field": "value"}}}],"time_field_name": "timestamp","start_time": 1620630000000,"end_time": 1621234800000,"min_time_interval": 86400000,"num_outputs": 10}
```

Reply:
The code snippet you provided (`if not isinstance(input_json, dict):`) does not directly check whether a given input is a valid JSON string or not. The isinstance() function in Python checks if an object is an instance of a particular class or data type. In this case, it checks if the input_json variable is not an instance of a dictionary (dict).

While a valid JSON string can be represented as a dictionary object in Python, this check alone does not guarantee that the input is a valid JSON string. It only verifies that the input_json is not a dictionary.

Reply:
Hmm, I will double check whether it accepts dictionaries.

Reply:
If the input is not an instance of a dictionary, json.loads will be called. Then, if it’s not a deserializable json, it will raise an exception. Otherwise, it returns a dictionary that is used in the query.